src/logic/pieces/Elephant.ts

```diff
@@ -0,0 +1,53 @@
+import Piece from "../Piece"
+import Board from "../Board"
+import Position from "../Position"
+
+/**
+ * A class to represent an Elephant in Chinese Chess.
+ */
+class Elephant extends Piece {
+  public constructor(code: number, position: Position) {
+    super(code, position)
+  }
+
+  public override getAllValidMoves(board: Board): Position[] {
+    const isRed: boolean = super.isRed()
+    const riverBoundary = isRed ? 5 : 4 // Red (y <= 4), Black (y >= 5)
+    const moveDirections: [number, number][] = [
+      [2, 2],
+      [2, -2],
+      [-2, 2],
+      [-2, -2]
+    ]
+
+    return moveDirections
+      .map(
+        ([dx, dy]) =>
```

Comment:
dx and dy in this case means delta x and delta y right?
Which means the difference or change in x or y variables

Reply:
nice! I like it.
I like this naming of these variables, it's concise and informative